python-hadoop/hadoop/io/BytesWritable.py

```diff
@@ -28,4 +28,4 @@ def readFields(self, data_input):
         self._value = data_input.readFully(size)
 
     def toString(self):
-        return ''.join(chr(x) for x in self._value)
+        return ''.join(chr(x) for x in self._value if x > 31 or x == 9)
```

Comment:
That is true, but I don't think there is a requirement that characters be printable. My understanding is that you could have a byte string of any kind. The mod 256 does seem to take care of the exceptions and is more general.